src/lib/dataviews/histogram/Histogram.ts

```diff
@@ -0,0 +1,39 @@
+import { Layer, Source } from '@/viz';
+import { BuiltInFilters } from '@/viz/filters/types';
+import { uuidv4 } from '@/core/utils/uuid';
+import { DataViewCalculation, HistogramDataViewOptions } from '../mode/DataViewMode';
+import { DataViewLocal } from '../mode/DataViewLocal';
+import { DataViewRemote } from '../mode/DataViewRemote';
+import { DataViewWrapper } from '../DataViewWrapper';
+import { HistogramImpl } from './HistogramImpl';
+
+export class HistogramDataView extends DataViewWrapper {
+  protected buildImpl(
+    dataSource: Layer | Source,
+    column: string,
+    options: HistogramDataViewOptions
+  ) {
+    let dataView;
+    const { mode } = options;
+
+    switch (mode) {
+      case DataViewCalculation.LOCAL: {
+        dataView = new DataViewLocal(dataSource as Layer, column);
+        break;
+      }
+
+      case DataViewCalculation.REMOTE: {
+        dataView = new DataViewRemote(dataSource as Source, column);
+        break;
+      }
+
+      default: {
+        dataView = new DataViewRemote(dataSource as Layer, column);
+        dataView.addFilter(`VIEWPORT_FILTER_${uuidv4()}`, BuiltInFilters.VIEWPORT);
+        break;
+      }
+    }
```

Comment:
@jesusbotella I think we will need here a similar approach as this one (still WIP) to make it work with GeoJSON, right? https://github.com/CartoDB/web-sdk/blob/193f57b518b6dd20af1e755369090a7fce24ca78/src/lib/dataviews/category/Category.ts

Reply:
Yes, kinda, we should simplify the code in there, but yes, we would need something like that.

Reply:
We will have that repetition also in Formula (@oleurud is going to add that to his PR). A candidate for further refactor in the future, but step by step